src/main/scala/com/fulcrumgenomics/bam/ZipperBams.scala

```diff
@@ -0,0 +1,191 @@
+    }
+
+    // Set the sort and group order
+    header.setSortOrder(unmapped.getSortOrder)
+    header.setGroupOrder(unmapped.getGroupOrder)
+    sort.foreach(s => s.applyTo(header))
+
+    header
+  }
+
+  /** Constructs a template iterator that assumes the input is sorted or grouped, and turns it into an async iter. */
+  def templateIterator(source: SamSource, bufferSize: Int): BetterBufferedIterator[Template] = {
+    val recs = source.iterator.bufferBetter
+    val templates = new Iterator[Template] {
+      override def hasNext: Boolean = recs.hasNext
+      override def next(): Template = Template(recs)
+    }
+
+    if (bufferSize > 0) {
+      AsyncIterator(templates, Some(bufferSize)).bufferBetter
+    }
+    else {
+      templates.bufferBetter
+    }
+  }
+
+  /** Merge information from the unmapped template into the mapped template. */
+  def merge(unmapped: Template, mapped: Template, tagInfo: TagInfo): Template = {
+    // Fix mate info first
+    mapped.fixMateInfo()
+
+    // Remove tags from the mapped records
+    for (rec <- mapped.allReads; tag <- tagInfo.remove) rec.remove(tag)
+
+    // Copy tags over from the unmapped record
+    for (u <- unmapped.primaryReads) {
+      // Get all the mapped records for the unmapped record
+      val ms = if (u.unpaired || u.firstOfPair) mapped.allR1s.toSeq else mapped.allR1s.toSeq
+      // Copy over tags on any positive strand reads first
+      ms.iterator.filter(_.positiveStrand).foreach(m => copyTags(u, m))
+
+      // Then reverse complement the unmapped read if we need to
+      if (tagInfo.hasRevsOrRevcomps && ms.exists(_.negativeStrand)) {
+        u.asSam.reverseComplement(tagInfo.setToRevcomp, tagInfo.setToReverse, true)
+      }
+
+      // And deal with the negative strand mappings
+      ms.iterator.filter(_.negativeStrand).foreach(m => copyTags(u, m))
+    }
+
+    mapped
+  }
+
+  /** Copies all tags from read to another. */
+  def copyTags(src: SamRecord, dest: SamRecord): Unit = {
+    src.attributes.foreach { case (tag, value) => dest(tag) = value }
+  }
+}
+
+@clp(group=ClpGroups.SamOrBam, description=
+  """
+    |Zips together an unmapped and mapped BAM to transfer metadata into the output BAM.
+    |
+    |Both the unmapped and mapped BAMs _must_ be a) queryname sorted or grouped (i.e. all records with the same
+    |name are grouped together in the file), and b) have the same ordering of querynames.  If either of these are
+    |violated the output is undefined!
+    |
+    |By default the mapped BAM is read from stdin and the output BAM is written to stdout. This can be changed
+    |using the `--input/-i` and `--output/-o` options.
+    |
+    |By default the output BAM file is emitted in the same order as the input BAMs.  This can be overridden
+    |using the `--sort` option, though in practice it may be faster to do the following:
+    |
+    |  `fgbio --compression 0 ZipperBams -i mapped.bam -u unmapped.bam -r ref.fa | samtools sort -@ 16`
```

Comment:
Also, why not use a triple-back-tick code block?

```suggestion
    |  `fgbio --compression 0 ZipperBams -i mapped.bam -u unmapped.bam -r ref.fa | samtools sort -@ $(nproc)`
```